PLUGINS.md

```diff
@@ -0,0 +1,264 @@
+# containerd plugins
+
+containerd supports extending its functionality using most of its defined
+interfaces. This includes using a customized runtime, snapshotter, content
+store, and even adding gRPC interfaces.
+
+## smart client model
+
+containerd has a smart client architecture, meaning any functionality which is
+not required by the daemon is done by the client. This includes most high
+level interactions such as creating a container's specification, interacting
+with an image registry, or loading an image from tar. Using containerd's Go
+client gives a users access to many points of extensions from creating their own
+options on container creation to resolving image registry names.
+
+See [containerd's Go documentation](https://godoc.org/github.com/containerd/containerd)
+
+## external plugins
+
+External plugins allow extending containerd's functionality using an officially
+released version of containerd without recompiling to add a plugin. containerd
+has two methods to use external plugins, either by making a binary available
+in containerd's PATH or configuring containerd to proxy to another gRPC service.
+
+### v2 runtimes
+
+The runtime v2 interface allows resolving runtimes to binaries on the system.
+These binaries are used to start the shim process for containerd and allows
+containerd to manage those containers using the runtime shim api returned by
+the binary.
+
+See [runtime v2 documentation](runtime/v2/README.md)
+
+### proxy plugins
+
+A proxy plugin is configured using containerd's config file and will be loaded
+alongside the internal plugins when containerd is started. These plugins are
+connected to containerd using a local socket serving one of containerd's gRPC
+API services. Each plugin is configured with a type and name just as internal
+plugins are.
+
+#### configuration
+
+Update the containerd config file, which defaults to
```

Comment:
may be me not being a native speaker, but `which defaults to` made me think that you would display the content of the file, not its location.

Reply:
That is a reasonable interpretation of the sentence, changing to `which by default is at` to be more clear.